packages/@lwc/engine/src/framework/services.ts

```diff
@@ -62,8 +62,8 @@ export function invokeServiceHook(vm: VM, cbs: ServiceCallback[]) {
             `Optimize invokeServiceHook() to be invoked only when needed`
         );
     }
-    const { component, data, def, context } = vm;
+    const { component, def, context } = vm;
     for (let i = 0, len = cbs.length; i < len; ++i) {
-        cbs[i].call(undefined, component, data, def, context);
+        cbs[i].call(undefined, component, {}, def, context);
```

Comment:
We no longer need to keep the VNodeData in the vm?

Reply:
If you look into the code the `data` property was never set. And since it doesn't fail any test, I am suspecting it is not used anymore.